diff --git a/glib/meson.build b/glib/meson.build
index 0fb6e2696..248f453cb 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -501,6 +501,8 @@ pkg.generate(libglib,
 )
 meson.override_dependency('glib-2.0', libglib_dep)
 
+if false
+
 # FIXME: This needs to be added to the `depends` of every invocation of
 # glib_compile_resources in the build process. We can drop this when
 # https://github.com/mesonbuild/meson/issues/13385 is fixed.
@@ -542,6 +544,8 @@ else
   meson.override_find_program('gtester', gtester)
 endif
 
+endif
+
 report_conf = configuration_data()
 report_conf.set('GLIB_VERSION', glib_version)
 report_conf.set('PYTHON', python_shebang)
diff --git a/meson.build b/meson.build
index c432f75ed..dc676ed40 100644
--- a/meson.build
+++ b/meson.build
@@ -2293,14 +2293,7 @@ else
   use_pcre2_static_flag = false
 endif
 
-# Import the gvdb sources as a subproject to avoid having the copylib in-tree
-subproject('gvdb', default_options: {'tests': false})
-gvdb_dep = dependency('gvdb')
-
 libm = cc.find_library('m', required : false)
-libffi_dep = dependency('libffi', version : '>= 3.0.0', default_options: {'werror': false})
-
-libz_dep = dependency('zlib')
 
 libatomic_test_code = '''
   int main (int argc, char ** argv) {
@@ -2731,12 +2724,6 @@ gnome = import('gnome')
 subdir('build-aux')
 subdir('tools')
 subdir('glib')
-subdir('gobject')
-subdir('gthread')
-subdir('gmodule')
-subdir('gio')
-subdir('girepository')
-subdir('fuzzing')
 if get_option('tests')
   subdir('tests')
 endif
@@ -2746,12 +2733,6 @@ if find_program('xgettext', required : get_option('nls')).found()
   subdir('po')
 endif
 
-# Install m4 macros that other projects use
-install_data('m4macros/glib-2.0.m4', 'm4macros/glib-gettext.m4', 'm4macros/gsettings.m4',
-  install_dir : get_option('datadir') / 'aclocal',
-  install_tag : 'devel',
-)
-
 # Check whether we support overriding the invalid parameter handler on Windows for _get_osfhandle(),
 # g_fsync() (i.e. _commit()), etc
 if host_system == 'windows'
@@ -2779,7 +2760,6 @@ endif
 rst2html5 = find_program('rst2html5', 'rst2html5.py', required: get_option('documentation'))
 
 gnome = import('gnome')
-subdir('docs/reference')
 
 summary({
   'host cpu' : host_machine.cpu_family(),
